DSharpPlus/Entities/Interaction/Application/DiscordAutoCompleteChoice.cs

```diff
@@ -34,46 +30,30 @@ private DiscordAutoCompleteChoice(string name)
     {
         if (name.Length is < 1 or > 100)
         {
-            throw new ArgumentException(
-                "Application command choice name cannot be empty or exceed 100 characters.",
-                nameof(name)
-            );
+            throw new ArgumentOutOfRangeException(nameof(name), "Application command choice name cannot be empty or exceed 100 characters.");
         }
 
         this.Name = name;
-        this.Value = null!;
     }
 
     /// <inheritdoc cref="DiscordAutoCompleteChoice(string)"/>
     /// <param name="name">The name of this option, which will be presented to the user.</param>
     /// <param name="value">The value of this option.</param>
-    public DiscordAutoCompleteChoice(string name, string value)
-        : this(name)
+    public DiscordAutoCompleteChoice(string name, object? value) : this(name) => this.Value = value switch
     {
-        if (value.Length > 100)
-        {
-            throw new ArgumentException(
-                "Application command choice value cannot exceed 100 characters.",
-                nameof(value)
-            );
-        }
-
-        this.Value = value;
+        string s => CheckStringValue(s),
+        int i => this.Value = i,
+        long l => this.Value = l,
+        double d => this.Value = d,
+        float f => this.Value = f,
+        null => null,
+        _ => throw new ArgumentException("Invalid value type.", nameof(value))
+    };

```

Comment:
code style: multi-line methods should not be expression-bodied, as per editorconfig
also, can we consider accepting more primitive types, casting to what we support?